src/auth/auth.controller.ts

```diff
@@ -0,0 +1,24 @@
+import { Controller, Body, Post, HttpCode, UseGuards } from '@nestjs/common';
+import { AuthService } from './auth.service';
+import { AuthDto } from './dto';
+import { LocalAuthGuard } from './guards/local-auth.guard';
+import { User } from '@prisma/client';
+import { AuthGuard } from '@nestjs/passport';
+
+@Controller('auth')
+export class AuthController {
+  constructor(private authService: AuthService) {}
+
+  @HttpCode(200)
+  @UseGuards(LocalAuthGuard)
+  @Post('signin')
+  async signIn(@Body() user: User) {
```

Comment:
include a response type too. Here `signIn` method is `async` and `this.authService.signIn` returns `Promise<{ access_token: string }>` -> the return type should be `Promise<{ access_token: string }>` so:

```suggestion
  async signIn(@Body() user: User): Promise<{ access_token: string }> {
```
but, I think I already suggested somewhere int this PR to move this type to a separate file so that we can do:
```suggestion
  async signIn(@Body() user: User): Promise<SomeExampleSignInResponseType> {
```

Reply:
Looking further into this, from `signIn` method you can return `String` for now instead of `{ access_token: string }` and here in controller map the response to a DTO response type. You should create a `SignInResponseDto` class like this:

```typescript
export class SignInResponseDto {
  constructor(public accessToken: string) {}

  public static from(accessToken: string): SignInResponseDto {
    return new SignInResponseDto(accessToken);
  }
}
```

and here in controller do:
```
const authServiceResult = await this.authService.signIn(user);

return SignInResponseDto.from(authServiceResult);
```